video/out/wayland_common.h

```diff
@@ -72,7 +72,6 @@ struct vo_wayland_state {
     struct mp_rect geometry;
     struct mp_rect window_size;
     float aspect_ratio;
-    bool fullscreen;
```

Comment:
I think we still need this one. `handle_toplevel_config` will check the surfaces for you, but something needs to keep track of its actual state.